Yosemite/Yosemite/Stores/OrderStore.swift

```diff
@@ -306,6 +306,29 @@ private extension OrderStore {
         }
     }
 
+    /// Updates, inserts, or prunes the provided StorageOrderItem's taxes using the provided read-only OrderItem
+    ///
+    private func handleOrderItemTaxes(_ readOnlyItem: Networking.OrderItem, _ storageItem: Storage.OrderItem, _ storage: StorageType) {
```

Comment:
How do I add the `handleOrderItemTaxes` to the other methods? If I add it to the `handleOrderItems`, it causes a crash.

Reply:
Just dumping here things as I go...

I modified handleOrderItems to look like this (added the call to handle taxes if there is an existing storage item:

```
    private func handleOrderItems(_ readOnlyOrder: Networking.Order, _ storageOrder: Storage.Order, _ storage: StorageType) {
        // Upsert the items from the read-only order
        for readOnlyItem in readOnlyOrder.items {
            if let existingStorageItem = storage.loadOrderItem(itemID: readOnlyItem.itemID) {
                handleOrderItemTaxes(readOnlyItem, existingStorageItem, storage)
                existingStorageItem.update(with: readOnlyItem)
            } else {
                let newStorageItem = storage.insertNewObject(ofType: Storage.OrderItem.self)
                newStorageItem.update(with: readOnlyItem)
                storageOrder.addToItems(newStorageItem)
            }
        }

        // Now, remove any objects that exist in storageOrder.items but not in readOnlyOrder.items
        storageOrder.items?.forEach { storageItem in
            if readOnlyOrder.items.first(where: { $0.itemID == storageItem.itemID } ) == nil {
                storageOrder.removeFromItems(storageItem)
                storage.deleteObject(storageItem)
            }
        }
    }
```

If I run the tests. `testSearchOrdersDoesNotProduceDuplicatedReferences` crashes because `[OrderItem addTaxObject:]: unrecognized selector sent to instance 0x7f90716bbde0`

Reply:
Looking into `OrderItem+CoreDataProperties` I don't see the accessors to add single OrderItemTax

Reply:
wait, it's there!
```
 @objc(addTaxObject:)
    @NSManaged public func addToTaxes(_ value: OrderItemTax)
```

Reply:
But the annotation does not match the conventions? 🤔 
If I update the annotation for the first two accessors:

```
extension OrderItem {

    @objc(addTaxesObject:)
    @NSManaged public func addToTaxes(_ value: OrderItemTax)

    @objc(removeTaxesObject:)
    @NSManaged public func removeFromTaxes(_ value: OrderItemTax)

    @objc(addTax:)
    @NSManaged public func addToTaxes(_ values: NSSet)

    @objc(removeTax:)
    @NSManaged public func removeFromTaxes(_ values: NSSet)

}
```

It does not crash anymore

Reply:
There are one test failing now: `testUpdateStoredOrderEffectivelyUpdatesPreexistantOrder` as the count of OrderItemTax does not match

Reply:
It seems like there might be something off in the logic in `handleOrderItemTaxes`. `po storageItem.taxes` in a breakpoint on line 326 sometimes prints an actual tax item, some others an empty collection, it seems like the tax items get removed somehow?

Reply:
Thank you Cesar! I know the unit tests aren't correct, but I wanted to get the crash figured out first.